moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -229,6 +229,14 @@ JointModelGroup::JointModelGroup(const std::string& group_name, const srdf::Mode
   {
     link_model_map_[link_model->getName()] = link_model;
     link_model_name_vector_.push_back(link_model->getName());
+    // if this is the first link of the group with a valid parent and includes geometry (for example `base_link`) it should included
+    if (link_model_with_geometry_vector_.empty() && link_model->getParentLinkModel() &&
```

Comment:
I'm not sure this is the correct approach. There can be more than one discrete subtree in a group, this would only consider the first one.

We discussed a similar issue I observed in https://github.com/ros-planning/moveit/issues/3570

I think the previous version of the code is correct for `is_chain_ == false`  (a list of `<link> or <joint>` tags in SRDF) but not for the case of `<chain>` where the root link is included but not parent joint.

Reply:
Good point, thanks for pointing that out! Do you think we should revert this change for now or just add a check for `is_chain_`?

Reply:
Hmm, this is getting surprisingly complicated ... `is_chain_` is not passed to the constructor but *guessed* about near the end of the constructor. They might be different things altogether?

According to the code in `RobotModel` a chain as specified in SRDF could also go from `finger_1_tip` to `finger_2_tip` with the root being somewhere in the middle (eg the `wrist`).

Maybe a list of links should also get passed in to the constructor.

I'm also not yet sure how this relates to collision checking and what the problem actually was.